### DISCUSSION TITLE
Certificate Signature Verification failed

### DISCUSSION THREAD

**Discussion by rendonvelez** [micromdm/micromdm#990] [Q&A]
Hello, I want to request your kind help to understand what is happening, even though I have followed the steps according to the documentation, I cannot generate the certificate for my server, these are the steps I am following:

1. I am running the service on my url
sudo micromdm serve -server-url=https://mdmserver.eatsyorders.com -api-key MypassWord -tls=false

2. mdmctl config set -name production -api-token MypassWord -server-url https://mdmserver.eatsyorders.com
3. mdmctl mdmcert vendor -password=MypassWord -country=US -email=myemail@mydomain.com
With the VendorCertificateRequest.csr generated in Apple Developer I generate a certificate for

Apple Push Notification service SSL (Sandbox & Production)
Establish connectivity between your notification server, the Apple Push Notification service sandbox, and production environments to deliver remote notifications to your app. When utilizing HTTP/2, the same certificate can be used to deliver app notifications, update ClockKit complication data, and alert background VoIP apps of incoming activity. A separate certificate is required for each app you distribute.

I download the generated aps.cer file

4. mdmctl mdmcert push -password=MypassWord -country=US -email=myemail@mydomain.com
5. mdmctl mdmcert vendor -sign -cert=./mdm-certificates/aps.cer -password=MypassWord

Now I load the plist generated at https://identity.apple.com/pushcert/ and the error is generated:

Certificate Signature Verification failed
Certificate Signature Verification failed because the [signature](https://www.apple.com/iphone/business/it/management.html) is invalid.

**Comment by jessepeterson** [comment]
It sounds like you're doing the right steps. Which version of MicroMDM are you using?

When you say "aps.cer" — is that the actual name? Usually it has a longer name named after the Developer Account. You don't have to share the name here, I'm just making sure you got the right certificate.

I'll also link you to the image we use for the APNs exchange from the [Certificate blog post](https://micromdm.io/blog/certificates/):
![APNs processs](https://micromdm.io/certificates/mdm_csr_process.png)

**Reply by rendonvelez** [reply]
Hello!

I am using version 1.12.1, the name asp.cer is the name with which the certificate produced by Apple Developer in step 3 is generated, which is generated using VendorCertificateRequest.csr, which is called mdm.cer in the instructions

**Reply by jessepeterson** [reply]
I see. Usually this is called `mdm.cer` I think. Did you create it using "MDM CSR" option in the Developer Portal? Check out video at the 35:55 mark: https://www.youtube.com/watch?v=WGKT-PyHz6I&start=2155

**Reply by rendonvelez** [reply]
Mmm, I see that I don't have that option in the generation of certificates

These are the options I have:

- Apple Push Notification service SSL (Sandbox)
- Apple Push Notification service SSL (Sandbox & Production)
- Pass Type ID Certificate
- Order Type ID Certificate
- Website Push ID Certificate
- Swift Package Collection Certificate
- Swift Package Certificate
- WatchKit Services Certificate
- VoIP Services Certificate
- Apple Pay Payment Processing Certificate
- Apple Pay Merchant Identity Certificate

**Reply by jessepeterson** [reply]
Okay, gotcha. This means your account does not have the "MDM CSR" option and you'll need to request it from Apple. You'll have to [contact Developer Relations](https://developer.apple.com/support/) and request the option be added to your developer account.
